ingestion-beam/src/test/java/com/mozilla/telemetry/DecoderMainTest.java

```diff
@@ -20,25 +27,39 @@
   @Rule
   public TemporaryFolder outputFolder = new TemporaryFolder();
 
-  private String outputPath;
-
-  @Before
-  public void initialize() {
-    outputPath = outputFolder.getRoot().getAbsolutePath();
-  }
+  @Rule
+  public final RedisServer redis = new RedisServer();
 
   @Test
   public void instantiateDecoderForCodeCoverage() {
     new Decoder();
   }
 
+  /** Make serialization of attributes map deterministic for these tests. */
+  @BeforeClass
+  public static void setUp() throws Exception {
+    Field mapperField = Json.class.getDeclaredField("MAPPER");
+    mapperField.setAccessible(true);
+    ObjectMapper mapper = (ObjectMapper) mapperField.get(null);
+    mapper.enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
+  }
+
+  /** Reset the ObjectMapper configurations we changed. */
+  @AfterClass
+  public static void tearDown() throws Exception {
+    Field mapperField = Json.class.getDeclaredField("MAPPER");
+    mapperField.setAccessible(true);
+    ObjectMapper mapper = (ObjectMapper) mapperField.get(null);
+    mapper.disable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
+  }
```

Comment:
This reflection business is a bit unfortunate, but I think it's good to keep the mapper private. In the future, it would be nice to build more robust testing utilities that can compare PubsubMessages rather than strings and could avoid this need.

Reply:
Wrote up https://github.com/mozilla/gcp-ingestion/issues/234